handlers/response/writer_test.go

```diff
@@ -0,0 +1,75 @@
+package response
+
+import (
+	. "github.com/smartystreets/goconvey/convey"
+	"testing"
+	"net/http/httptest"
+	"net/http"
+	"encoding/json"
+)
+
+type child struct {
+	Name string `json:"value"`
+}
+
+type parent struct {
+	Name  string `json:"name"`
+	Child child `json:"child"`
+}
+
+type mockOnsJsonEncoder struct {
+	invocationArgs  []interface{}
+	mockedBehaviour func(w http.ResponseWriter, value interface{}, status int) error
+}
+
+func mockEncoderInit() *mockOnsJsonEncoder {
+	impl := onsJSONEncoder{}
+	return &mockOnsJsonEncoder{invocationArgs: make([]interface{}, 0), mockedBehaviour: impl.writeResponseJSON}
+}
+
+func (mock *mockOnsJsonEncoder) writeResponseJSON(w http.ResponseWriter, value interface{}, status int) error {
+	mock.invocationArgs = append(mock.invocationArgs, value)
+	return mock.mockedBehaviour(w, value, status)
+}
+
+func TestWriteJSON(t *testing.T) {
+
+	Convey("Should Write value to response as JSON, set the expected http response status code & set the content " +
+		"type header with the expected value", t, func() {
```

Comment:
Consider splitting up the long single statment into multiple ''Convey" calls to improve output format.